knitr/planetary_motion/planetary_motion.rmd

```diff
@@ -0,0 +1,735 @@
+Unfortunately we cannot compute the likelihood across a 7-dimensional grid and we wouldn't quite know how to visualize it.
+Our proposition is to look at _conditional likelihoods_, that is fix some of the parameters and examine the remaining ones.
+This, from a certain point of view, very much amounts to studying a simplification of our model.
+To begin, we fix all parameters (based on the correct values or equivalently estimates from the well-behaving Markov chains), except for $q_*^x$.
+
+```{r, fig.height=5}
+star_x <- seq(from = -0.5, to = 0.8, by = 0.01)
+star_s <- array(NA, dim = c(length(star_x), 2))
+star_s[, 1] <- star_x
+star_s[, 2] <- rep(0, length(star_x))
+
+k <- 1
+q0 <- c(1.0, 0)
+p0 <- c(0, 1.0)
+dt <- 0.001
+m <- 1
+n_obs <- nrow(q_obs)
+
+lk <- rep(NA, nrow(star_s))
+for (i in 1:nrow(star_s)) {
+  q_sim <- solve_trajectory(q0, p0, dt, k, m, n_obs, ts, star_s[i, ])
+  
+  lk[i] <- 
+    sum(dnorm(q_obs[, 1], q_sim[, 1], sigma_x, log = T)) +
+    sum(dnorm(q_obs[, 2], q_sim[, 2], sigma_y, log = T))
+}
+
+plot <- ggplot(data = data.frame(star_x = star_x, lk = lk),
+               aes(x = star_x, y = lk)) + theme_bw() +
+  geom_line() + theme(text = element_text(size = 18)) +
+  ylab("Conditional log likelihood") + xlab(TeX("$q_*^x$"))
+plot
+```
+
+This is the type of profile we expect.
+We can extend this to a heat map, with both coordinates of $q_*$ varying, and observe "ripples" that suggest the presence of local modes.
+```{r, fig.height=3}
+star_x <- seq(from = -0.5, to = 0.8, by = 0.05)
+star_y <- seq(from = -0.5, to = 0.5, by = 0.05)
+star_s <- array(NA, c(length(star_x), length(star_y), 2))
+
+for (i in 1:length(star_x)) {
+  for (j in 1:length(star_y)) star_s[i, j, ] = c(star_x[i], star_y[j])
+}
+
+star_data <- array(NA, c(length(star_s) / 2, 2))
+for (i in 1:length(star_x)) {
+  for (j in 1:length(star_y)) {
+    index <- (j - 1) * length(star_x) + i
+    star_data[index, ] <- star_s[i, j, ]
+  }
+}
+
+lk <- rep(NA, nrow(star_data)) 
+for (i in 1:nrow(star_data)) {
+  q_sim <- solve_trajectory(q0, p0, dt, k, m, n_obs, ts, star_data[i, ])
+  
+  lk[i] <- 
+    sum(dnorm(q_obs[, 1], q_sim[, 1], sigma_x, log = T)) +
+    sum(dnorm(q_obs[, 2], q_sim[, 2], sigma_y, log = T))
+}
+
+plot2 <- ggplot(data = data.frame(star_x = star_data[, 1],
+                                  star_y = star_data[, 2],
+                                  lk = lk), 
+                aes(x = star_x, y = star_y, fill = lk)) +
+  geom_tile() + theme_bw() + theme(text = element_text(size = 18)) +
+  xlab(TeX("$q_*^x$")) + ylab(TeX("$q_*^y$")) + labs(fill = "log likelihood")
+plot2
+```
+
+These figures support our conjecture and, along with the log posterior and the posterior predictive checks, suggest the modes are mathematical artefacts, much like to ones we have previously probed.
+Let us caution that the behavior of the posterior density can change quite a bit as we add more moving parts.
```

Comment:
Good point. The reason I'm being careful here is because the figures support our conjecture, but they do not prove it. But using the word "support" is probably good enough.